woocommerce/woocommerce-android
# Handle back when sw update in progress

Parent issue #3957 

This PR 
- Replaces circular progress indicator with a progress bar showing the current progress
- Displays a warning and a cancel button when the user presses back

Known issues:
Stripe SDK contains [a bug](https://github.com/stripe/stripe-terminal-android/issues/119) which prevents us from cancelling an ongoing progress. If the user clicks on cancel but starts the update flow again it'll always fail with "UNEXPECTED_ERROR.UNEXPECTED_SDK_ERROR: Operation in progress wasn't completed before starting new operation". This should be fixed in v2 of the SDK and it's an edge case anyway, so I don't think it's worth investigating.

<img src="https://user-images.githubusercontent.com/2261188/124877024-f0b95e00-dfca-11eb-8209-fca2041d2152.gif" width="200px" />

To test
1. Connect to a reader
2. Leave the "update available" dialog by pressing back
3. Tap on "Update Reader's software"
4. Tap on "update"
5. Press back and notice a warning and a cancel button is shown
6. Tap on cancel and notice the update dialog is dismissed 

Update release notes:
